### PR TITLE
fix: radio button storybook bug

### LINKED ISSUE #15705: [Selectable Components: Radio Button] Dev Implementation
## Acceptance criteria

- [x] [Enhancement] To be updated in storybook - by default, no option will be selected
image

--- 

- [x] [Bug] Toggling on the `defaultChecked`, it didn't show up

Screen Recording 2024-02-07 at 11 53 07 AM

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #15705 

- added the  `defaultSelected` prop in the `RadioButton` playground
- updated docs to better support the need for `RadioButton` (s) to be rendered as children of `RadioButtonGroup`
#### Changelog

**Removed**

Removed the following props from `RadioButton`

- checked 
- defaultChecked

#### Testing / Reviewing
- Checkout the `defaultSelected` prop in the `RadioButton` playground

### CODE REVIEW

File: packages/react/src/components/RadioButton/RadioButton.stories.js [hunk 1]
```diff
@@ -108,13 +120,21 @@ export const Playground = (args) => {
...
   helperText: 'Helper text',
   invalidText: 'Invalid selection',
   warn: false,
   warnText: 'Please notice the warning',
 };
 
 Playground.argTypes = {
+  defaultSelected: {
+    description: 'Specify the `<RadioButton>` to be selected by default',
+    options: ['radio-1', 'radio-2','radio-3'],
```
Comment (andreancardona) [inline packages/react/src/components/RadioButton/RadioButton.stories.js:133]:
```suggestion
    options: ['radio-1', 'radio-2',' radio-3'],
```

Comment (andreancardona) [inline packages/react/src/components/RadioButton/RadioButton.stories.js]:
```suggestion
    options: ['radio-1', 'radio-2', 'radio-3'],
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Looking at the issue, looks like we also need to remove the `defaultSelected` prop from the default story.

**Review by tay1orjones — APPROVED** [review]
Yeah, I agree with Alison. Other than that I think this looks great

**Comment by andreancardona** [comment]
it shouldn't be selected? Just want to clarify here:

**Review by alisonjoseph — APPROVED** [review]

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
oops hit approve accidentally. The issue says not selected. I think if you just remove `defaultChecked` from the story that should work. 

Screenshot 2024-03-29 at 8 26 11 AM

**Review by Kritvi-bhatia17 — DISMISSED** [review]
Hi @andreancardona, I noticed the smart adjustment you made for the default dropdown options!
However, I'm encountering an issue where the default dropdown options 1, 2, and 3 aren't functioning as expected. When I select different options, the output consistently shows the 3rd option as selected. Could you please take a look?

https://github.com/carbon-design-system/carbon/assets/158570656/c0ea6aa6-5511-43e6-8b6a-55852115c662

**Comment by tay1orjones** [comment]
@Kritvi-bhatia17 `defaultSelected` only impacts which one is selected in the initial render. To see your change you'll need to refresh the story


https://github.com/carbon-design-system/carbon/assets/3360588/b427e00b-2a69-4ae9-be64-e3fe135ac2ec

**Comment by Kritvi-bhatia17** [comment]
> @Kritvi-bhatia17 `defaultSelected` only impacts which one is selected in the initial render. To see your change you'll need to refresh the story
> 
>  defaults.update.after.refresh.mov

Ohh damn, I didn't know that earlier. Thanks Taylor!

**Review by Kritvi-bhatia17 — APPROVED** [review]
Nice work @andreancardona 🔥
Looks good to me from design perspective!

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**